lib/iaas/AzureClient.js

```diff
@@ -0,0 +1,192 @@
+const BaseCloudClient = require('./BaseCloudClient');
+const NotFound = errors.NotFound;
+const Unauthorized = errors.Unauthorized;
+const Forbidden = errors.Forbidden;
+
+class AzureClient extends BaseCloudClient {
+  constructor(settings) {
+    super(settings);
+    this.storage = this.constructor.createStorageClient(_
+      .chain(this.settings)
+      .omit('name')
+      .set('provider', this.provider)
+      .value()
+    );
+    this.computeClient = this.constructor.createComputeClient(_
+      .chain(this.settings)
+      .set('provider', this.provider)
+      .value()
+    );
+    Promise.promisifyAll(this.computeClient.snapshots);
+
+    this.storage.on('log::*', function onmessage(message, obj) {
+      const event = _.nth(this.event.split('::'), 1);
+      const level = _.includes([
+        'warn',
+        'info',
+        'verbose',
+        'debug'
+      ], event) ? event : 'debug';
+      if (obj) {
+        logger.log(level, message, obj);
+      } else {
+        logger.log(level, message);
+      }
+    });
+  }
+
+  getContainer(container) {
+    if (arguments.length < 1) {
+      container = this.containerName;
+    }
+    return this.storage
+      .getContainerPropertiesAsync(container)
+      .then(result => result);
+  }
+
+  list(container, options) {
+    if (arguments.length < 2) {
+      options = container;
+      container = this.containerName;
+    }
+    const prefix = options ? options.prefix : null;
+    //TODO Need to update following line with value from 'options' e.g. options.marker
+    let continuationToken = null;
+    //TODO : maxResult is 5000 
+    //https://azure.github.io/azure-storage-node/BlobService.html#listBlobsSegmentedWithPrefix__anchor
+    //If want to fetch more use 'continuationToken' returned in result
+    return this.storage  
+      .listBlobsSegmentedWithPrefixAsync(container, prefix, continuationToken)  
+      .then(result => {
+        const listOfFiles = result.entries;
+        continuationToken = result.continuationToken;
+        let isTruncated = continuationToken ? true : false;         
+        const files = [];     
+        _.each(listOfFiles, file => files.push(_       
+          .chain(file)       
+          .pick('name', 'lastModified')       
+          .set('isTruncated', isTruncated)       
+          .value()          
+        ));      
+        return files;    
+      });
+  }
+
+  remove(container, file) {
+    if (arguments.length < 2) {
+      file = container;
+      container = this.containerName;
+    }
+    logger.debug(`Deleting file ${file} in container ${container} `);
+    return this.storage
+      .deleteBlobAsync(container, file)
+      .then(response => logger.info(`Response of blob deletion from cloud storage for ${file} :`, response))
+      .catch(BaseCloudClient.providerErrorTypes.Unauthorized, err => {
+        logger.error(err.message);
+        throw new Unauthorized('Authorization at the cloud storage provider failed');
```

Comment:
Better to send `err.message` to this constructor instead of this hard coded message. There is anyways a hardcoded message in errors.js for this class in case its input as undefined.

Reply:
err.message only gives 'Unauthorized'. I'll put a reference of blob name and container in the statement.

Reply:
updated the message